cli/tests/cfncluster-unittest.py

```diff
@@ -0,0 +1,295 @@
+        success_message = 'INFO:cfncluster.cfncluster:MasterPublicIP:'
+        error_prefix = "CRITICAL:"
+        self.assertTrue(success_message in log)
+        self.assertFalse(error_prefix in log)
+
+    @mock_ec2
+    @mock_cloudformation
+    @mock_s3
+    def test_cfn_cluster_create_fail(self):
+        template_url = setup_configurations()
+        args = CreateClusterArgs('', True)
+        with self.assertRaises(SystemExit) as sys_ex:
+            cfncluster.create(args)
+
+        self.assertEqual(sys_ex.exception.code, 1)
+        log = test_log_stream.getvalue()
+        error_prefix = "CRITICAL:"
+        self.assertTrue(error_prefix in log)
+
+    @mock_ec2
+    @mock_cloudformation
+    @mock_s3
+    def test_cfn_cluster_list_empty(self):
+        args = BaseArgs()
+        cfncluster.list(args)
+        self.assertEqual(test_log_stream.tell(), 0)
+
+    @mock_ec2
+    @mock_cloudformation
+    @mock_s3
+    def test_cfn_cluster_list_nonempty(self):
+        template_url = setup_configurations()
+        args = CreateClusterArgs(template_url, False)
+        cfncluster.create(args)
+        cfncluster.list(args)
+        log = test_log_stream.getvalue()
+        self.assertTrue(args.cluster_name in log)
+
+    @mock_ec2
+    @mock_cloudformation
+    @mock_autoscaling
+    @mock_s3
+    def test_cfn_cluster_update_no_reset(self):
+        template_url = setup_configurations()
+        args = UpdateClusterArgs(template_url, True, False)
+        cfncluster.create(args)
+        cfncluster.update(args)
+        success_message = 'INFO:cfncluster.cfncluster:Status: UPDATE_COMPLETE'
+        log = test_log_stream.getvalue()
+        self.assertTrue(success_message in log)
+        error_prefix = "CRITICAL:"
+        self.assertFalse(error_prefix in log)
+
+    @mock_ec2
+    @mock_cloudformation
+    @mock_autoscaling
+    @mock_s3
+    def test_cfn_cluster_update_with_reset(self):
+        template_url = setup_configurations()
+        args = UpdateClusterArgs(template_url, True, True)
+        cfncluster.create(args)
+        cfncluster.update(args)
+        success_message = 'INFO:cfncluster.cfncluster:Status: UPDATE_COMPLETE'
+        log = test_log_stream.getvalue()
+        self.assertTrue(success_message in log)
+        error_prefix = "CRITICAL:"
+        self.assertFalse(error_prefix in log)
+
+    @mock_ec2
+    @mock_cloudformation
+    @mock_autoscaling
+    @mock_s3
+    def test_cfn_cluster_update_fail(self):
+        template_url = setup_configurations()
+        args = UpdateClusterArgs(template_url, True, False)
+        with self.assertRaises(SystemExit) as sys_ex:
+            cfncluster.update(args)
+        self.assertEqual(sys_ex.exception.code, 1)
+        log = test_log_stream.getvalue()
+        error_prefix = "CRITICAL:"
+        self.assertTrue(error_prefix in log)
+
+    @mock_ec2
+    @mock_cloudformation
+    @mock_s3
+    @mock_autoscaling
+    def test_cfn_cluster_delete(self):
```

Comment:
can you also add a delete unit test for when the cluster doesn't exist?

Reply:
Added